src/client.cpp

```diff
@@ -269,10 +269,9 @@ int waybar::Client::main(int argc, char *argv[]) {
   gtk_app->hold();
   gtk_app->run();
   bars.clear();
-  zxdg_output_manager_v1_destroy(xdg_output_manager);
-  zwlr_layer_shell_v1_destroy(layer_shell);
-  zwp_idle_inhibit_manager_v1_destroy(idle_inhibit_manager);
-  wl_registry_destroy(registry);
-  wl_display_disconnect(wl_display);
```

Comment:
`Gdk::Display::get_default()` is a singleton, and since `wl_display` is retrieved from it, is (and the related components here) are also singletons. This means that if we destroy them we can't recreate them, so in order to execute `Client::main()` multiple times the simplest solution is to just not destroy them.

Since the functions return the same objects every time, this does not result in a memory leak.